src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -26,7 +26,7 @@ class PythonSetup(Subsystem):
   def register_options(cls, register):
     super(PythonSetup, cls).register_options(register)
     register('--interpreter-constraints', advanced=True, default=['CPython>=2.7,<3', 'CPython>=3.6,<4'], type=list,
-             metavar='<requirement>',
+             metavar='<requirement>', fingerprint=True,
```

Comment:
Should this be removed? It works without it, so I would err on the side of yes, remove it for a smaller diff?

Reply:
Could you describe why this continues to work without it? I am not sure I understand that and I worry that if the implementation changes in the future we will assume something like `fingerprint=True` is set and then be surprised when it's not.

Reply:
It still passes because we do not rely on the result of `self.fingerprint` in creating the `pantsd` file that stores the cache. The cache filename is solely based off of the result of `invalidated(targets).all_vts`:

https://github.com/pantsbuild/pants/blob/36fdbd8973763e8b2dae994b25dd19f79cf20301/src/python/pants/backend/python/tasks/select_interpreter.py#L90-L97
 
Since we ignore `self.fingerprint` in determining the file name, it makes no difference whether we set `fingerprint=True` here.

--

Note this wasn't immediately obvious, because we _do_ use `self.fingerprint` to setup the `cache_key_generator` used by `invalidated()`:

https://github.com/pantsbuild/pants/blob/36fdbd8973763e8b2dae994b25dd19f79cf20301/src/python/pants/task/task.py#L497-L508

But my experiment shows it does not actually change the end hash id.

1. Apply this diff
```diff
diff --git a/src/python/pants/backend/python/tasks/select_interpreter.py b/src/python/pants/backend/python/tasks/select_interpreter.py
index 76b58d406..029afa7c1 100644
--- a/src/python/pants/backend/python/tasks/select_interpreter.py
+++ b/src/python/pants/backend/python/tasks/select_interpreter.py
@@ -77,6 +77,7 @@ class SelectInterpreter(Task):
         if not invalidation_check.all_vts else
         VersionedTargetSet.from_versioned_targets(invalidation_check.all_vts).cache_key.hash
       )
+      raise ValueError(target_set_id)
       interpreter_path_file = self._interpreter_path_file(target_set_id)
       interpreter = self._get_interpreter(interpreter_path_file, python_tgts)
```
2. Run `PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="['CPython==3.7.*']" PANTS_RESOLVER_USE_MANYLINUX=true ./pants test --cache-ignore tests/python/pants_test/util:strutil`
3. Run `PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="['CPython==3.7.*']" PANTS_RESOLVER_USE_MANYLINUX=false ./pants test --cache-ignore tests/python/pants_test/util:strutil`

 
cc @benjyw

Reply:
> I worry that if the implementation changes in the future we will assume something like fingerprint=True is set and then be surprised when it's not.

Worth noting that we now have a dedicated unit test for this invalidation behavior, so hopefully this would never happen.

Reply:
Since these values are incorporated into your fingerprint strategy, I believe that you would technically be OK to remove the flag in this case. But it is better to over-fingerprint than it is to leave comments explaining why it is safe to under fingerprint.